Portal/vip-api/src/main/java/fr/insalyon/creatis/vip/api/rest/controller/processing/ExecutionControler.java

```diff
@@ -59,91 +59,70 @@
-    }
-
-    private ExecutionBusiness buildExecutionBusiness(ApiContext apiContext,
-                                                     PipelineBusiness pipelineBusiness) {
-        if (pipelineBusiness == null) {pipelineBusiness = new PipelineBusiness(apiContext, workflowBusiness,
-                applicationBusiness, classBusiness);
-        }
-        return new ExecutionBusiness(apiContext,
-                simulationBusiness, workflowBusiness, configurationBusiness, applicationBusiness,
-                pipelineBusiness, transferPoolBusiness);
+    @RequestMapping
+    public Execution[] listExecutions(
+            @RequestParam(required = false) String studyIdentifier,
+            @RequestParam(required = false) Integer offset,
+            @RequestParam(required = false) Integer limit
+    ) throws ApiException {
+        ApiUtils.methodInvocationLog("listExecutions", studyIdentifier, offset, limit);
+        restApiBusiness.getApiContext(httpServletRequest, true);
+        if (studyIdentifier != null) throw new ApiException("studyIdentifier not supportet yet");
+        if (offset != null) throw new ApiException("offset not supported yet");
+        int executionMaxNb = environment.getProperty(DEFAULT_LIMIT_LIST_EXECUTION, Integer.class);
+        if (limit == null) limit = executionMaxNb;
+        if (limit > executionMaxNb) throw new ApiException("limit parameter too high");
+        return executionBusiness.listExecutions(limit);
     }
 
-    @RequestMapping
-    public Execution[] listExecutions() throws ApiException {
-        ApiUtils.methodInvocationLog("listExecutions");
-        ApiContext apiContext = restApiBusiness.getApiContext(httpServletRequest, true);
-        ExecutionBusiness eb = buildExecutionBusiness(apiContext);
-        return eb.listExecutions(environment.getProperty(DEFAULT_LIMIT_LIST_EXECUTION, Integer.class));
+    @RequestMapping(value = "count", produces = "text/plain;charset=UTF-8")
+    public String countExecutions(
+            @RequestParam(required = false) String studyIdentifier
+    ) throws ApiException {
+        ApiUtils.methodInvocationLog("countExecutions");
+        restApiBusiness.getApiContext(httpServletRequest, true);
+        if (studyIdentifier != null) throw new ApiException("studyIdentifier not supportet yet");
+        return String.valueOf(executionBusiness.countExecutions());
     }
 
     @RequestMapping("/{executionId}")
     public Execution getExecution(@PathVariable String executionId) throws ApiException {
         ApiUtils.methodInvocationLog("getExecution", executionId);
-        ApiContext apiContext = restApiBusiness.getApiContext(httpServletRequest, true);
-        ExecutionBusiness eb = buildExecutionBusiness(apiContext);
-        eb.checkIfUserCanAccessExecution(executionId);
-        return eb.getExecution(executionId,false);
+        restApiBusiness.getApiContext(httpServletRequest, true);
+        executionBusiness.checkIfUserCanAccessExecution(executionId);
```

Comment:
Do you think it would be useful to move the checks inside getExecution, as for the DataController methods?